src/jacobians.jl

```diff
@@ -120,7 +120,7 @@ end
 
 
 function _make_Ji(::AbstractArray, xtype, dx, color_i, nrows, ncols)
-    Ji = mapreduce(i -> i==color_i ? dx : zeros(xtype,nrows), hcat, 1:ncols)
+    Ji = mapreduce(i -> i==color_i ? dx : (ret = similar(dx, nrows); fill!(ret, 0.)), hcat, 1:ncols)
```

Comment:
```suggestion
    Ji = mapreduce(i -> i==color_i ? dx : (ret = similar(dx, nrows); fill!(ret, zero(eltype(dx)))), hcat, 1:ncols)
```

Reply:
It still shouldn't use similar and fill in the out of place setup,

Reply:
Is there a `zero(::AbstractArray, dims...)` function in ArrayInterface.jl? If not, we should add it.

Reply:
There isn't, but yes we should add it.

Reply:
Should we overload `Base.zero` or make a new function like `similar_zero`?

Reply:
I think we just overload `Base.zero` since it doesn't have any other meaning.